spec/app/domain/util/log_message_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'Util::LogMessageClass' do
+  context 'object' do
+    let(:log_message) { "some log message" }
+    subject(:log_message_class) { Util::LogMessageClass.new(log_message) }
+
+    it 'has the expected message' do
+      expect(log_message_class.new.to_s).to eq(log_message)
+    end
+  end
+
+  context 'templating' do
+    let(:log_message_template) { "Variable is \#{0}." }
+
+    subject(:log_message_class) { Util::LogMessageClass.new(log_message_template) }
+
+    it 'works for nil' do
+      expect(log_message_class.new(nil).to_s).to eq("Variable is #nil.")
+    end
+
+    it 'works for strings' do
+      expect(log_message_class.new("").to_s).to eq("Variable is #.")
+      expect(log_message_class.new("ABC").to_s).to eq("Variable is #ABC.")
+    end
+
+    it 'works for Fixnums' do
+      expect(log_message_class.new(9).to_s).to eq("Variable is #9.")
+      expect(log_message_class.new(0).to_s).to eq("Variable is #0.")
+    end
+
+    it 'works for booleans' do
+      expect(log_message_class.new(false).to_s).to eq("Variable is #false.")
+      expect(log_message_class.new(true).to_s).to eq("Variable is #true.")
+    end
+
+    it 'works for objects' do
+      expect(log_message_class.new(Class).to_s).to eq("Variable is #Class.")
+      expect(log_message_class.new(Util::LogMessageClass).to_s).to eq("Variable is #Util::LogMessageClass.")
+    end
+
+    it 'works for arrays' do
+      expect(log_message_class.new(["aA", "bB"]).to_s).to eq("Variable is #[\"aA\", \"bB\"].")
+    end
+
+    it 'handles overflow args' do
+      expect(log_message_class.new("AA", "BB").to_s).to eq("Variable is #AA.")
+    end
+
+    it 'handles underflow args' do
+      expect(log_message_class.new.to_s).to eq("Variable is #\{0\}.")
+    end
```

Comment:
Seems like a missing template variable should error.

Reply:
wanted to maintain the behaviour of error_class